fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/service/DelinquencyReadPlatformServiceImpl.java

```diff
@@ -109,52 +111,30 @@ public Collection<LoanDelinquencyTagHistoryData> retrieveDelinquencyRangeHistory
 
     @Override
     public CollectionData calculateLoanCollectionData(final Long loanId) {
-        final LocalDate businessDate = DateUtils.getBusinessLocalDate();
         final Optional<Loan> optLoan = this.loanRepository.findById(loanId);
 
         CollectionData collectionData = CollectionData.template();
         if (optLoan.isPresent()) {
+            final LocalDate businessDate = DateUtils.getBusinessLocalDate();
             final Loan loan = optLoan.get();
+            Integer graceDays = 0;
+            if (loan.loanProduct().getLoanProductConfigurableAttributes().getGraceOnArrearsAgingBoolean()) {
+                graceDays = loan.loanProduct().getLoanProductRelatedDetail().getGraceOnArrearsAgeing();
+                if (graceDays == null) {
+                    graceDays = 0;
+                }
+            }
 
-            collectionData.setAvailableDisbursementAmount(loan.getApprovedPrincipal().subtract(loan.getDisbursedAmount()));
+            collectionData.setAvailableDisbursementAmount(loan.getApprovedPrincipal().subtract(loan.getNetDisbursalAmount()));
```

Comment:
i dont think this is correct.

The netDisbursalAmount will be overriden always by the very latest disbursement amount.
**Example:**
- Approved amount is 1000
- I do a disbursement with amount of 400
- The netDisbursalAmount will be 400, so it will set the available disbursement amount is 600
- If I do a 2nd disbursement with amount of 600
- The netDisbursalAmount will be 600, so the available disbursement amount will be set to 400, but that is not correct. We should consider all of the disbursement details to get the correct amount.

@josehernandezfintecheandomx what do you think?

Reply:
@josehernandezfintecheandomx This is still not fixed. Please share your thoughts about the issue I listed above.

Reply:
Updared